BricklinkSharp.Client/IBricklinkRequestHandler.cs

```diff
@@ -34,6 +34,6 @@ namespace BricklinkSharp.Client
 {
     public interface IBricklinkRequestHandler
     {
-        Task OnRequestAsync(string url, CancellationToken ct);
+        Task OnRequestAsync(string url, BricklinkApiRequestTypes requestType, CancellationToken ct);
```

Comment:
Do we need url anymore?
I think it makes sense to also include the Request method (GET, POST, PUT or DELETE) also as custom enums?

There is another issue I came accross: the OnRequestAsync interface method must also contain the BricklinkCredentials that are used to make the request, otherwise the whole feature renders useless for users who pass Credentials on a request-basis (including myself) ;D

Reply:
we can skip the urls, alright.
http verbs are fine, yes. Will do.
I can add the credentials but are you sure that it makes sense to ship also those keys and secrets through the application so many times? Can we limit the credentials? e.g. using the consumerkey or username?

Reply:
I dont think that passing the credentials to the callback will be problem (they are passed by Reference anyway). I definitely require the credentials for my server scenario.

Reply:
alright, expect this addition later today.
In the meanwhile, this is the result:

![grafik](https://github.com/gebirgslok/BricklinkSharp/assets/2912080/a77fa2a6-5778-4c20-9fd8-573f8f094ccf)

works like a charm! I will aggregate the information a bit further (Item, Color, Order, ...) for easier displaying